spec/acceptance/resolved_spec.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+require 'spec_helper_acceptance'
+
+describe 'systemd' do
+  context 'configure systemd resolved' do
+    it 'works idempotently with no errors' do
+      pp = <<-PUPPET
+      class{'systemd':
+        manage_resolved    => true,
+        manage_resolv_conf => #{default[:hypervisor] != 'docker'},
+      }
+      PUPPET
+      apply_manifest(pp, catch_failures: true)
+      # RedHat 9 and newer installs package first run before fact  $facts['internal_services'] is set
+      apply_manifest(pp, catch_failures: true) if Gem::Version.new(fact('os.release.major')) >= Gem::Version.new('9') && (fact('os.family') == 'RedHat')
```

Comment:
This would be more explicit. Not insisting on it, but it could help the reader?
```suggestion
      separate_resolved_package = Gem::Version.new(fact('os.release.major')) >= Gem::Version.new('9') && (fact('os.family') == 'RedHat')
      apply_manifest(pp, catch_failures: true) if separate_resolved_package
```

Reply:
With the comment I think I prefer, seems more obvious what to delete when C8 is dead.